### PR TITLE
Remove with(out)-system-key tests

### PR DESCRIPTION
**Pull request by jaymode**
This change removes the variants of the rolling upgrade and full
cluster restart tests that use or do not use a system key. These tests
were added during 5.x when the system key was still used for security
and now the system key is only used as the watcher encryption key so
duplicating rolling upgrade and full cluster restarts is not needed.

The change here removes the subprojects for testing these scenarios and
defaults to always run with the watcher sensitive values encrypted for
these tests.

### CODE REVIEW

**Review by mark-vieira — CHANGES_REQUESTED** [review]
Couple minor things, but overall LGTM.

FWIW, it would be helpful in the future to push formatting changes in a separate commit. As it is it makes reviewing this PR pretty difficult as it's hard to separate formatting changes from actual functional changes.

File: x-pack/qa/full-cluster-restart/build.gradle [hunk 1]
```diff
@@ -90,223 +97,159 @@ forbiddenPatterns {
...
-  }
-
-  licenseHeaders {
-    approvedLicenses << 'Apache'
-  }
-
-  forbiddenPatterns {
-    exclude '**/system_key'
-  }
+testingConventions.enabled = false
```
[the remark this answers is not in the document]
Comment (jaymode) [reply x-pack/qa/full-cluster-restart/build.gradle]: Good catch

**Comment by jaymode** [comment]
Sorry about the formatting changes. This is a result of copying a section and pasting it in intellij. Intellij wants to indent with 4 spaces while most build files are indented with 2 for some reason.

**Review by mark-vieira — APPROVED** [review]
👍

File: x-pack/qa/full-cluster-restart/build.gradle [hunk 2]
```diff
@@ -3,12 +3,18 @@ import org.elasticsearch.gradle.test.RestIntegTestTask
...
 
 // Apply the java plugin to this project so the sources can be edited in an IDE
-apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.standalone-test'
+
 unitTest.enabled = false
 
+repositories {
+    maven {
+        url "https://snapshots.elastic.co/maven"
```
Comment (rjernst) [inline x-pack/qa/full-cluster-restart/build.gradle]: I'm confused why this is necessary. The build shouldn't be using any snapshots; it should all be local. Additionally, this was using the official release url before, not just snapshots?

**Review by rjernst — COMMENTED** [review]
LGTM, one question

**Review by albertzaharovits — APPROVED** [review]
Thanks for the ping! LGTM, but I can just grok these scripts, If CI is happy so am I.

File: x-pack/qa/full-cluster-restart/build.gradle (hunk 2 above)
Comment (jaymode) [reply x-pack/qa/full-cluster-restart/build.gradle]: @mark-vieira pointed out that the standalone-test plugin already adds this repo. You're right that this shouldn't be using a repo, I'll remove it. I am guessing that this was a leftover item from when we didn't do that...

**Pull request merged**